### DISCUSSION TITLE
Read and write files

### DISCUSSION THREAD

**Discussion by Gal-Peled** [jupyterlite/jupyterlite#790] [Q&A]
I am trying to overwrite a text file and I cannot. Am I doing somthing wrong ? I am a bit new to Python. 

```Python

# create a new txt file with 'Hellow World'
with open('text_data.txt', 'w+') as f:
    f.write("Hellow World")
    f.seek(0)            
    myText1 = f.read()

# clear the file content 
open('text_data.txt', 'w').close()
    
with open('text_data.txt', 'r') as f:
    myText2 = f.read()

print(f'{myText1=}')
print(f'{myText2=}')
```
myText1='Hellow World'
myText2='Hellow World'

I expect to get an empty string in myText2: 
myText1='Hellow World'
myText2=''

**Comment by oscar6echo** [comment]
Some info:

If is simpler to delete a file rather than clearing its content.  
Using [pathlib](https://docs.python.org/3/library/pathlib.html):

```py
filename = 'text_data.txt'
content = 'Hello World'
content2 = 'AB'

p = Path(filename)
p.write_text(content)
p.unlink()
p.write_text(content2)
p.read_text()
# output
# 'AB'
```

Besides, beware when you modify the contents of a file.  
The behavior is not very intuitive. So it is recommended to experiment.  
Example:

```py
filename = 'text_data.txt'
content = 'Hello World'
content2 = 'AB'

p = Path(filename)
p.write_text(content)
p.write_text(content2)
p.read_text()
# output
# 'ABllo World'
```
Even if you explicitly truncate a file, it does not seem to work.  
So the safe way is to delete and create (or append to) files but not amend them:

```py
filename = 'text_data.txt'
content = 'Hello World'

p = Path(filename)
p.write_text(content)
with open(p, 'w') as f:
    f.truncate(0)
p.read_text()
# output
# 'Hello World'
```

**Reply by Gal-Peled** [reply]
thx , you are correct I also trried to use truncate. Yet I think it should work per the python documentation.
